### ISSUE TITLE
Feature requset: Give instructors a way to resend 'result published' link to selected students

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#6634]
Currently, instructors can resend join invites and submission link emails to selected students, in case a student did not receive the automated email.
Similarly, an instructor should be able to resend the 'results published' email to selected students.

**Comment by HirdayGupta** [comment]
Working on this.

**Comment by HirdayGupta** [comment]
Dropping a note that this issue is taking longer than expected. I have managed to get the button and action working to remind all the students, however, I underestimated the "Selected students" part of the issue.

I am currently working on modifying the code for `remindParticularStudentsTag` to abstract it out as simply a modal with a student list checkbox. And then I will work on adding a new action to send to specific students only.

**Comment by HirdayGupta** [comment]
> I am currently working on modifying the code for remindParticularStudentsTag to abstract it out as simply a modal with a student list checkbox.

@damithc can I do this as a separate PR? I feel that there may need to be some discussion regarding this and it may unnecessarily add to the diffs and review time of this PR.

**Comment by damithc** [comment]
> @damithc can I do this as a separate PR? I feel that there may need to be some discussion regarding this and it may unnecessarily add to the diffs and review time of this PR.

Sure.

**Comment by damithc** [comment]
Increasing priority because this is an important feature that is missing.

**Comment by bqnguyen94** [comment]
@damithc do we need a way to resend the link to all students as well, like the remind submission feature?

**Comment by damithc** [comment]
> @damithc do we need a way to resend the link to all students as well, like the remind submission feature?

Assuming you are planning to present the user with a list of students to choose from (similar to the `remind particular students` feature), we can give a `select all` option in case the instructor wishes to send the link to all.

**Comment by bqnguyen94** [comment]
> Assuming you are planning to present the user with a list of students to choose from (similar to the remind particular students feature), we can give a select all option in case the instructor wishes to send the link to all.

For `remind particular students` feature, `send to all students` is on the home page and student list is presented in a separated modal. Therefore for `resend result published email` I'm thinking maybe the `send to all students` option can be on the home page as well instead of putting it in the modal for consistency's sake?

**Comment by damithc** [comment]
BTW, the link `Resend link to view results` can be given in the `Results` menu.

**Comment by damithc** [comment]
> For remind particular students feature, send to all students is on the home page and student list is presented in a separated modal. Therefore for resend result published email I'm thinking maybe the send to all students option can be on the home page as well instead of putting it in the modal for consistency's sake?

Those two links too should be collapsed into one. That modal can give the following options. 
- [ ] `all submitted`
- [ ] `all yet to submit`

**Comment by bqnguyen94** [comment]
> Those two links too should be collapsed into one. That modal can give the following options.

@damithc this is for `remind particular students` only right?

**Comment by damithc** [comment]
> @damithc this is for remind particular students only right?

Yes, but with this we can remove `Remind all students` entirely. I've created a new issue for this #8530
